lib/Core/Memory.cpp

```diff
@@ -624,44 +628,92 @@ void ObjectState::print() const {
 
 /***/
 
-PersistentState::PersistentState(const ObjectState *os, const Array *cacheLines)
+PersistentState::PersistentState(const ObjectState *os, 
+                                 const Array *cacheLines,
+                                 const Array *rootCauses,
+                                 const Array *idxArray)
   : ObjectState(*os),
-    cacheLineUpdates(0, 0),
-    pendingCacheLineUpdates(0, 0) {
-  cacheLineUpdates = UpdateList(cacheLines, 0);
-  pendingCacheLineUpdates = UpdateList(cacheLineUpdates);
+    cacheLineUpdates(cacheLines, nullptr),
+    pendingCacheLineUpdates(cacheLineUpdates),
+    idxEmptyUpdates(idxArray, nullptr),
+    rootCauseLocations(rootCauses, nullptr) {
+  // -- For cacheline stuff
+  // 0 <= X <= 2^32
```

Comment:
Tiny thing, but should it not be 0 <= X < 2^32, rather than "<= 2^32"?

Reply:
yup.